src/content/getusermedia/getdisplaymedia/js/main.js

```diff
@@ -6,40 +6,127 @@
  *  tree.
  */
 'use strict';
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element@0.6.2?module';
 
-// Polyfill in Firefox.
-// See https://blog.mozilla.org/webrtc/getdisplaymedia-now-available-in-adapter-js/
-if (adapter.browserDetails.browser == 'firefox') {
-  adapter.browserShim.shimGetDisplayMedia(window, 'screen');
-}
+class ScreenSharing extends LitElement {
+  constructor() {
+    super();
+    this.downloadUrl = null;
+    this.buttons = {
+      startCapturing: true,
+      stopCapturing: false,
+      downloadRecording: false,
+    };
+    this.stream = null;
+    this.recording = [];
+    this.mediaRecorder = null;
+  }
 
-function handleSuccess(stream) {
-  const video = document.querySelector('video');
-  video.srcObject = stream;
+  static get properties() {
+    return {
+      buttons: Object,
```

Comment:
this wont work in Lit... has to use type: Object... but then Object is a useless serializer

Reply:
Yeah, this was the best I could come up with. I think I'll clean it up with three separate booleans instead.